src/client.js

```diff
@@ -707,6 +711,9 @@ class Client {
                 if (that._traceAPICalls)
                     console.log(`SOAP//failure ${ex.toString()}`);
                 that._notifyObservers((observer) => observer.onSOAPCallFailure && observer.onSOAPCallFailure(soapCall, ex) );
+                // Call session expiration callback in case of 403
+                if (ex.statusCode == 403)
```

Comment:
Why? 403 is forbidden. It may not be a session expiration. Would there not be an infinite loop in this xase?

Reply:
I'm not sure of the logic here.
The callback is asynchronous, shouldn't there be an await here?
Are we retrying the failed call after a failure?